lib/3scale/backend/stats/delete_job_def.rb

```diff
@@ -0,0 +1,65 @@
+module ThreeScale
+  module Backend
+    module Stats
+      class DeleteJobDef
+        ATTRIBUTES = %i[service_id applications metrics users from to context_info].freeze
+        private_constant :ATTRIBUTES
+        attr_reader(*ATTRIBUTES)
+
+        def self.attribute_names
+          ATTRIBUTES
+        end
+
+        def initialize(params = {})
+          ATTRIBUTES.each do |key|
+            instance_variable_set("@#{key}".to_sym, params[key]) unless params[key].nil?
+          end
+          validate
+        end
+
+        def run_async
+          Resque.enqueue(PartitionGeneratorJob, Time.now.getutc.to_f, service_id, applications,
+                         metrics, users, from, to, context_info)
+        end
+
+        def to_json
+          to_hash.to_json
+        end
+
+        def to_hash
+          Hash[ATTRIBUTES.collect { |key| [key, send(key)] }]
+        end
+
+        private
+
+        def validate
```

Comment:
I forgot to mention that we should probably put some kind of limit on the `to` parameter or on the time interval between `to` and `from`. Otherwise, someone could send a call to delete all the keys from 1970 up until now or something like that. We can fix this in the integration branch later.